src/umpire/ResourceManager.cpp

```diff
@@ -565,7 +565,10 @@ ResourceManager::reallocate(void* src_ptr, std::size_t size)
 
   void* dst_ptr = nullptr;
 
-  if (!src_ptr) {
+  if (!src_ptr || size == 0) {
+    if (src_ptr)
+      m_default_allocator->deallocate(src_ptr);
+
     dst_ptr = m_default_allocator->allocate(size);
   } else {
     auto& op_registry = op::MemoryOperationRegistry::getInstance();
```

Comment:
Thanks @mcfadden8!

I think there is still an issue with this ``else`` branch.

* Line 566 sets ``dest_ptr = nullptr``.
* Line 587 dereferences this null pointer

```cpp
   void* dst_ptr = nullptr; 
   ...
    op->transform(src_ptr, &dst_ptr, alloc_record, alloc_record, size);
```

Note: This is the code path that my function in axom is taking.

Reply:
As discussed in person, I will go ahead and resolve this conversation as it is working now.

Reply:
Thanks for the updates @mcfadden8 -- I verified that your changes work in my application.